packages/eslint-plugin/tests/rules/strict-boolean-expressions.test.ts

```diff
@@ -516,29 +516,6 @@ declare const nullableString: string | null;
 assert(3 as any, nullableString);
       `,
     },
-    // Intentional TS error - A rest parameter must be last in a parameter list.
```

Comment:
These tests were for coverage reasons that no longer make sense with the new algorithm.
None of them crashes. 2 of them now give a false positive, but they match how TS thinks about the situation 🤷‍♂️